packages/core/src/amazonqDoc/session/sessionState.ts

```diff
@@ -49,344 +21,120 @@ import {
+            }
+            case codegenResult.codeGenerationStatusDetail?.includes('PROMPT_REFUSAL'): {
+                return new PromptRefusalException(codegenResult.codeGenerationRemainingIterationCount || 0)
+            }
+            case codegenResult.codeGenerationStatusDetail?.includes('Guardrails'): {
+                return new DocServiceError(i18n('AWS.amazonq.doc.error.docGen.default'), 'GuardrailsException')
+            }
+            case codegenResult.codeGenerationStatusDetail?.includes('EmptyPatch'): {
+                if (codegenResult.codeGenerationStatusDetail?.includes('NO_CHANGE_REQUIRED')) {
+                    return new NoChangeRequiredException()
                 }
-                await this.config.proxyClient.startCodeGeneration(
-                    this.config.conversationId,
-                    this.config.uploadId,
-                    action.msg,
-                    Intent.DOC,
-                    codeGenerationId,
-                    undefined,
-                    action.folderPath ? { documentation: { type: 'README', scope: action.folderPath } } : undefined
-                )
-
-                const codeGeneration = await this.generateCode({
-                    messenger: action.messenger,
-                    fs: action.fs,
-                    codeGenerationId,
-                    telemetry: action.telemetry,
-                    workspaceFolders: this.config.workspaceFolders,
-                    mode: action.mode,
+                return new DocServiceError(i18n('AWS.amazonq.doc.error.docGen.default'), 'EmptyPatchException')
+            }
+            case codegenResult.codeGenerationStatusDetail?.includes('Throttling'): {
+                return new DocServiceError(i18n('AWS.amazonq.featureDev.error.throttling'), 'ThrottlingException')
+            }
+            default: {
+                return new ToolkitError(i18n('AWS.amazonq.doc.error.docGen.default'), {
+                    code: 'DocGenerationFailed',
                 })
+            }
+        }
+    }
 
-                if (codeGeneration && !action.tokenSource?.token.isCancellationRequested) {
-                    this.config.currentCodeGenerationId = codeGenerationId
-                    this.currentCodeGenerationId = codeGenerationId
-                }
-
-                this.filePaths = codeGeneration.newFiles
-                this.deletedFiles = codeGeneration.deletedFiles
-                this.references = codeGeneration.references
-                this.codeGenerationRemainingIterationCount = codeGeneration.codeGenerationRemainingIterationCount
-                this.codeGenerationTotalIterationCount = codeGeneration.codeGenerationTotalIterationCount
+    protected async startCodeGeneration(action: SessionStateAction, codeGenerationId: string): Promise<void> {
+        if (!action.tokenSource?.token.isCancellationRequested) {
+            ;(action.messenger as DocMessenger).sendDocProgress(
```

Comment:
I don't think we need this casting here
```suggestion
            action.messenger.sendDocProgress(
```